src/DynamoCoreWpf/Utilities/ResourceUtilities.cs

```diff
@@ -601,13 +602,21 @@ internal static async void LoadWebBrowser(HtmlPage htmlPage, WebView2 webBrowser
                     };
                 }
 
-                await webBrowserComponent.EnsureCoreWebView2Async();
-                // Context menu disabled
-                webBrowserComponent.CoreWebView2.Settings.AreDefaultContextMenusEnabled = false;
-                webBrowserComponent.NavigateToString(bodyHtmlPage);
+                try
+                {
+                    await webBrowserComponent.Initialize();
+
+                    // Context menu disabled
+                    webBrowserComponent.CoreWebView2.Settings.AreDefaultContextMenusEnabled = false;
+                    webBrowserComponent.NavigateToString(bodyHtmlPage);
 
-                webBrowserComponent.CoreWebView2.Settings.IsZoomControlEnabled = false;
-                webBrowserComponent.CoreWebView2.Settings.IsPinchZoomEnabled = false;
+                    webBrowserComponent.CoreWebView2.Settings.IsZoomControlEnabled = false;
+                    webBrowserComponent.CoreWebView2.Settings.IsPinchZoomEnabled = false;
+                }
+                catch (ObjectDisposedException ex)
```

Comment:
How can this exception be thrown now that we are waiting for initialize to happen?

Reply:
It can be thrown here https://github.com/DynamoDS/Dynamo/pull/14888/files#diff-405bba4d4ce51bef98ee2ea6e37e396b6834800587417bd4ccfa855590099ef3R43

we throw that exception so that callers of Initialize() can short-circuit their initialization code in case the object was disposed before Initialize finished

Reply:
I guess I'm trying to understand what will be the case when it's disposed while it's being initialized at the same time. Are you saying these things can happen concurrently in different threads?

Reply:
I can happen on the same thread, but asynchronously 
For example a test:
```
using (webview2View = new Webview2View){
   webview2View.Initialize();// async method might take a couple of seconds
} // might exit block before webview2View.Initialize finishes (dispose will be called here)
```

Reply:
Ah ok, so these are additional guardrails for sloppy testing code.